### ISSUE TITLE
Autoplay does not work for Require inventory and Conditionals tutorial

### ISSUE THREAD

**Issue by xsebek** [swarm-game/swarm#1424]
**Describe the bug**

When I run
```
swarm -i scenarios/Tutorials/requireinv.yaml --autoplay
swarm -i scenarios/Tutorials/conditionals.yaml --autoplay
```

The code is not properly run and instead in the REPL it shows this:
```
it0 : unit = <a1>
```

When I input the definitions manually and run the code, it works fine though.

**Expected behavior**

Autoplay should work for all scenarios that have solutions.

**Screenshots**

<img width="627" alt="Screenshot 2023-08-12 at 14 09 49" src="https://github.com/swarm-game/swarm/assets/44544735/dc4505fd-1d97-4291-8172-f04c19b65927">

**Additional context**

It would be good to know _why_ this is not an issue in integration tests, but does not work with autoplay.

**Comment by xsebek** [comment]
I had to require a `logger` in the solution to see what its problem was:
```bash
❯ curl -s localhost:5357/robot/1 | yq '._robotLog.[]._leText' -P | sed 's/^/# /g'
# You do not have the devices required for:
#   'ishere'
#   Please equip:
#   - scanner
```

So it turns out to be a bug with command requirements propagation.

@byorgey do you know how can this possibly work in integration tests, but not with autoplay? 😕

**Comment by byorgey** [comment]
This is strange, since the `base` program does not use `ishere` at all.  It is the `condition` program which uses `ishere`, which should be run by a system robot.

**Comment by byorgey** [comment]
I still don't know exactly what the problem was, but it looks like this is fixed by #1928.
